lib/messagebird.js

```diff
@@ -395,6 +395,308 @@ module.exports = function (accessKey, timeout) {
           httpRequest('POST', '/lookup/' + phoneNumber + '/hlr', params, callback);
         }
       }
+    },
+
+    contacts: {
+
+      /**
+       * Create a new contact. Params is optional.
+       *
+       * @param {String} phoneNumber
+       * @param {Object} params
+       * @param {Function} callback
+       * @return void
+       */
+      create: function (phoneNumber, params, callback) {
+        if (typeof params === 'function') {
+          callback = params;
+          params = {};
+        }
+
+        params.msisdn = phoneNumber;
+
+        httpRequest('POST', '/contacts', params, callback);
+      },
+
+      /**
+       * Deletes an existing contact. The callback is invoked with an error if
+       * applicable, but the data will never contain anything meaningful as the
+       * API returns an empty response for successful deletes.
+       *
+       * @param {String} id
+       * @param {Function} callback
+       * @return void
+       */
+      delete: function (id, callback) {
+        httpRequest('DELETE', '/contacts/' + id, callback);
+      },
+
+      /**
+       * Lists existing contacts. Pagination is optional. If a limit is set, an
+       * offset is also required.
+       *
+       * @param {Number} limit
+       * @param {Number} offset
+       * @param {Function} callback
+       * @return void
+       */
+      list: function (limit, offset, callback) {
+        var params = null;
+
+        if (typeof callback === 'function') {
+          params = {
+            limit: limit,
+            offset: offset
```

Comment:
Given the strict requirement of both `limit` and `offset`, perhaps we could set some defaults?

```js
{
  limit: limit || defaultLimit,
  offset: offset || defaultOffset
}
```

Reply:
To be consistent with other "listing" endpoints in the client, we support `list(callback)` and `list(limit, offset, callback)`.
If we don't provide any, the API will fall back to defaults on their side (offset 0, limit 25 IIRC). I think it's ok to rely on that: otherwise we'd get some pretty ugly code. Doing the following would override  the callback when no pagination is set:

```javascript
limit = limit || defaultLimit;
offset = offset || defaultOffset;
```